tidb-lightning/tidb-lightning-log-message-explanation.md

```diff
@@ -0,0 +1,317 @@
+As part of precheck, Lightning uses the first data file of each table to check if source data file and target cluster table schema are matched.
+
+```
+[INFO] [version.go:360] ["detect server version"] [type=TiDB] [version=5.4.0]
+```
+[version.go:360](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/version/version.go#L360):
+Detect and print the current TiDB server version. To import data in local backend mode, TiDB with version higher than 4.0 is required.
+
+We also need to check server version for [detecting data confilcts](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/version/version.go#L224).
+
+```
+[INFO] [check_info.go:995] ["sample file start"] [table=sbtest1]
+```
+[check_info.go:995](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/check_info.go#L995): As part of precheck, it estimates source data size to determine: 
+- [the local disk has enough space if Lighting is in local backend mode](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/check_info.go#L462); 
+- [the target cluster has enough space to store transformed kv pairs](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/check_info.go#L102). 
+
+It calculates the file size vs. kv pairs size ratio by sampling the first source data file of each table, and using the ratio times source data file size to estimate the size of transformed kv pairs.
+
+```
+[INFO] [check_info.go:1080] ["Sample source data"] [table=sbtest1] [IndexRatio=1.3037832180660969] [IsSourceOrder=true]  
+```
+[check_info.go:1080](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/check_info.go#L1080): Table source file size vs. kv pairs size ratio has been calculated.
+
+```
+[INFO] [pd.go:415] ["pause scheduler successful at beginning"] [name="[balance-region-scheduler,balance-leader-scheduler,balance-hot-region-scheduler]"] 
+[INFO] [pd.go:423] ["pause configs successful at beginning"] [cfg="{\"enable-location-replacement\":\"false\",\"leader-schedule-limit\":4,\"max-merge-region-keys\":0,\"max-merge-region-size\":0,\"max-pending-peer-count\":2147483647,\"max-snapshot-count\":40,\"region-schedule-limit\":40}"]
+```
+
+[pd.go:415](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/pdutil/pd.go#L415), [pd.go:423](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/pdutil/pd.go#L423): In local backend mode, some [pd schedulers](https://docs.pingcap.com/tidb/stable/tidb-scheduling) are disabled and some [config settings are changed](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/pdutil/pd.go#L417) to split and scatter tikv regions and ingest ssts.
+
+```
+[INFO] [restore.go:1683] ["switch to import mode"]
+```
+[restore.go:1683](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/restore.go#L1683):
+In local backend mode, it turns each TiKV node into import mode to speed up import process, but sacrifices its storage space. If it uses tidb backend mode, it does not need to switch TiKV to import mode.
```

Comment:
maybe we can link to "import mode" in https://docs.pingcap.com/tidb/stable/tidb-lightning-glossary#import-mode